src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -804,13 +810,19 @@ namespace winrt::TerminalApp::implementation
     // - Creates a new tab with the given settings. If the tab bar is not being
     //      currently displayed, it will be shown.
     // Arguments:
+    // - profileGuid: ID to use to lookup profile settings for this connection
     // - settings: the TerminalSettings object to use to create the TerminalControl with.
-    void TerminalPage::_CreateNewTabFromSettings(GUID profileGuid, TerminalApp::TerminalSettings settings)
+    // - existingConnection: optionally receives a connection from the outside world instead of attempting to create one
+    void TerminalPage::_CreateNewTabFromSettings(GUID profileGuid, TerminalApp::TerminalSettings settings, TerminalConnection::ITerminalConnection existingConnection)
     {
         // Initialize the new tab
+        auto connection = existingConnection;
```

Comment:
```suggestion
        const auto connection = existingConnection ? existingConnection : _CreateConnectionFromSettings(profileGuid, settings);
```

This may be a way to sneak a `const` in there.

Reply:
Can't actually const it because the debug tap std::tie below assigns to it. But I did make it a ternary anyway.